File: peptidematch_web/docs/src/example_3.py
import urllib,urllib2

url = 'http://alanine.bioinformatics.udel.edu/peptidematch_new/webservices/peptidematch_rest'

params = {
'format':'tab',
'query':'DPETERQ',
'organism':'5888,537012'
}

data = urllib.urlencode(params)
request = urllib2.Request(url, data)
request.add_header('User-Agent', 'Python contact')
response = urllib2.urlopen(request)
page = response.read(200000)
print page

File: peptidematch_web/docs/src/example_3.pl
use strict;
use warnings;
use LWP::UserAgent;

my $base = 'http://alanine.bioinformatics.udel.edu/peptidematch_new/webservices/peptidematch_rest';

my $params = {
  format => 'tab',
  query => 'DPETERQ',
  organism => '5888,537012'
};

my $agent = LWP::UserAgent->new(agent => "libwww-perl");
push @{$agent->requests_redirectable}, 'POST';

my $response = $agent->post("$base", $params);

while (my $wait = $response->header('Retry-After')) {
  print STDERR "Waiting ($wait)...\n";
  sleep $wait;
  $response = $agent->get($response->base);
}

$response->is_success ?
  print $response->content :
  die 'Failed, got ' . $response->status_line .
    ' for ' . $response->request->uri . "\n";

